dss/api/bundles/__init__.py

```diff
@@ -115,28 +113,102 @@ def put(uuid: str, replica: str, json_request_body: dict, version: str=None):
         BundleMetadata.CREATOR_UID: json_request_body['creator_uid'],
     }
 
-    # if it already exists, then it's a failure.
-    try:
-        handle.get_user_metadata(bucket, bundle_manifest_object_name)
-    except BlobNotFoundError:
-        # write manifest to persistent store
-        handle.upload_file_handle(
+    was_created = _save_object_if_dne(
+        handle,
+        bucket,
+        bundle_manifest_object_name,
+        bundle_metadata,
+        "bundle_already_exists",
+        f"bundle with UUID {uuid} and version {version} already exists"
+    )
+    status_code = requests.codes.created if was_created else requests.codes.ok
+
+    return jsonify(dict(version=version)), status_code
+
+
+@dss_handler
+def delete(uuid: str, replica: str, json_request_body: dict, version: str = None):
+    owner = request.token_info['email']
+
+    if owner not in ADMIN_USER_EMAILS:
+        raise DSSException(
+            requests.codes.forbidden,
+            "Forbidden",
+            "You can't delete bundles with these credentials!",
+        )
+
+    uuid = uuid.lower()
+    version = datetime_to_version_format(iso8601.parse_date(version)) if version else None
+
+    bundle_prefix = _format_bundle_prefix(uuid, version)
+    tombstone_object_name = bundle_prefix + '.dead'
+    tombstone_object_data = _create_tombstone_data(
+        creator_uid=json_request_body['creator_uid'],
+        version=version,
+        reason=json_request_body.get('reason'),
+    )
+
+    handle, hca_handle, bucket = Config.get_cloud_specific_handles(replica)
+
+    if test_object_exists(handle, bucket, bundle_prefix, test_type='prefix'):
```

Comment:
I would not do this as it implies the only suffix that might ever be created is `.dead`.  What if we add something else in the future?